packages/core/src/App/Components/Layout/Footer/__tests__/network-status.spec.tsx

```diff
@@ -1,56 +1,48 @@
-// TODO refactor old tests in this component
 import React from 'react';
-import { Popover } from '@deriv/components';
-import { NetworkStatus } from '../network-status.jsx';
+import userEvent from '@testing-library/user-event';
 import { render, screen } from '@testing-library/react';
+import { NetworkStatus } from '../network-status.jsx';
+
+const MockNetworkStatus = ({ is_mobile = true }) => (
+    <NetworkStatus status={{ class: 'offline' }} is_mobile={is_mobile} />
+);
+
+describe('network-status component', () => {
+    it('should has "network-status__wrapper--is-mobile" class when the "is_mobile" property is true', () => {
+        render(<MockNetworkStatus />);
+        const div_element = screen.getByTestId('dt_network_status');
+        expect(div_element).toHaveClass('network-status__wrapper--is-mobile');
+    });
 
-const status = {
-    class: 'online',
-    tooltip: 'Online',
-};
+    it('should has correct class based on class passed in the "status" property', () => {
+        const { rerender } = render(<NetworkStatus status={{ class: 'offline' }} />);
+        expect(screen.getByTestId('dt_network_status_element')).toHaveClass('network-status__circle--offline');
 
-describe('NetworkStatus Component', () => {
-    it('should has "network-status__wrapper--is-mobile" class when the "is_mobile" is true', () => {
-        render(<NetworkStatus status={status} is_mobile />);
-        const divElement = screen.getByTestId('dt_network_status_id');
-        expect(divElement).toHaveClass('network-status__wrapper--is-mobile');
+        rerender(<NetworkStatus status={{ class: 'online' }} />);
```

Comment:
why do you need to run rerender on this?

Reply:
Thank u @ali-hosseini-fs for your revision
I'm passing different props in each render because of that I rerender the component